File: FundamentosDeLaProgramacionLuisJoyanes4taEdicion/chapter06/JAVA/FactorialFunction.java
/*Libro:Fundamentos de Programación, Algoritmos, estructura de datos y objetos 4ta edición. 
    Capitulo 06.-Subprogramas (subalgoritmos): Funciones
    Problema 04.-Diseñar un algoritmo que contenga un subprograma de cálculo del factorial de un número y una llamada al
                mismo.
    
    Ejemplo4
    Pagina 207
*/
import java.util.Scanner;

public class FactorialFunction{
    static int showFactorial(int x){
        if(x<=1){
            return 1;
        }else{
            return x*showFactorial(x-1);
        }

    }

    public static void main(String[] args){
        Scanner in=new Scanner(System.in);
        int x=0;
        System.out.print("Programa que calcula un factorial:\nIngrese un numero: ");
        x=in.nextInt();

        System.out.print("\nResultado del factorial:\n"+x+"! = "+showFactorial(x)+"\n\n");
    }
}

File: FundamentosDeLaProgramacionLuisJoyanes4taEdicion/chapter06/JAVASCRIPT/JS/factorialFunction.js
/*Libro:Fundamentos de Programación, Algoritmos, estructura de datos y objetos 4ta edición. 
    Capitulo 06.-Subprogramas (subalgoritmos): Funciones
    Problema 04.-Diseñar un algoritmo que contenga un subprograma de cálculo del factorial de un número y una llamada al
                mismo.
    
    Ejemplo4
    Pagina 207
*/
function calculate(){
    let x=0;

    if(document.getElementById("idBtnCalculate").value=="Calcular"){  
        x=parseInt(document.getElementById("idX").value);

        document.getElementById("idResult").value=showFactorial(x);
        document.getElementById("idBtnCalculate").value="Limpiar";
    }else{
        cleanPage();
    }
}

function showFactorial(x){
    if(x<=1){
        return 1;
    }else{
        return x*showFactorial(x-1);
    }

}

function cleanPage(){
    document.getElementById("idX").value="";
    document.getElementById("idResult").value="";
    document.getElementById("idBtnCalculate").value="Calcular";
}